src/character.cpp

```diff
@@ -207,43 +207,55 @@ double Character::aim_per_move( const item& gun, double recoil ) const
         if( mod->sight_dispersion < 0 || mod->aim_cost <= 0 ) {
             continue; // skip gunmods which don't provide a sight
         }
-        if( effective_dispersion( mod->sight_dispersion ) < recoil && mod->aim_cost < cost ) {
-            cost  = mod->aim_cost;
+        if( effective_dispersion( mod->sight_dispersion ) < recoil && mod->aim_cost < sight_cost ) {
+            sight_cost  = mod->aim_cost;
             limit = effective_dispersion( mod->sight_dispersion );
         }
     }
 
-    if( cost == INT_MAX ) {
-        return 0; // no suitable sights (already at maxium aim)
+    if( sight_cost == INT_MAX ) {
+        // No suitable sights (already at maxium aim).
+        return 0;
     }
 
-    // each 5 points (combined) of hand encumbrance increases aim cost by one unit
-    cost += round ( ( encumb( bp_hand_l ) + encumb( bp_hand_r ) ) / 10.0 );
-
-    /** @EFFECT_DEX increases aiming speed */
-    cost += 8 - dex_cur;
+    // Overal strategy for determining aim speed is to sum the factors that contribute to it,
+    // then scale that speed by current recoil level.
+    // Player capabilities make aiming faster, and aim speed slows down as it approaches 0.
+    // Base speed is non-zero to prevent extreme rate changes as aim speed approaches 0.
+    double aim_speed = 10.0;
 
+    // Ranges [0 - 10]
     /** @EFFECT_PISTOL increases aiming speed for pistols */
     /** @EFFECT_SMG increases aiming speed for SMGs */
     /** @EFFECT_RIFLE increases aiming speed for rifles */
     /** @EFFECT_SHOTGUN increases aiming speed for shotguns */
     /** @EFFECT_LAUNCHER increases aiming speed for launchers */
-    cost += ( ( MAX_SKILL / 2 ) - get_skill_level( gun.gun_skill() ) ) * 2;
+    aim_speed += std::min( MAX_SKILL, static_cast<int>( get_skill_level( gun.gun_skill() ) ) );
+
+    // Range [0 - 12]
+    /** @EFFECT_DEX increases aiming speed */
+    aim_speed += std::max( 0, get_dex() - 8 );
```

Comment:
Design smell: threshold.
But this one can be trivially un-thresholded, by moving that `-8` to `aim_speed` and just adding `get_dex` here.